app/screens/more_channels/more_channels.js

```diff
@@ -201,13 +202,20 @@ class MoreChannels extends PureComponent {
     };
 
     onSelectChannel = async (id) => {
+        const {actions, currentTeamId, currentUserId} = this.props;
+        const {channels} = this.state;
+
         this.emitCanCreateChannel(false);
         this.setState({adding: true});
-        await this.props.actions.joinChannel(
-            this.props.currentUserId,
-            this.props.currentTeamId,
-            id);
-        await this.props.actions.handleSelectChannel(id);
+        await actions.joinChannel(currentUserId, currentTeamId, id);
```

Comment:
I'm fine with that as long as we make sure we get back to it soon. The bug I'm dealing with right now is caused by this exact issue: it fails to join a team, ignores the error, and then the app gets stuck in an unusable state